src/neo4j_graphrag/tool.py

```diff
@@ -0,0 +1,224 @@
+    """Integer parameter for tools."""
+
+    def __init__(
+        self,
+        description: str,
+        required: bool = False,
+        minimum: Optional[int] = None,
+        maximum: Optional[int] = None,
+    ):
+        super().__init__(description, required)
+        self.minimum = minimum
+        self.maximum = maximum
+
+    def to_dict(self) -> Dict[str, Any]:
+        result: Dict[str, Any] = {
+            "type": ParameterType.INTEGER,
+            "description": self.description,
+        }
+        if self.minimum is not None:
+            result["minimum"] = self.minimum
+        if self.maximum is not None:
+            result["maximum"] = self.maximum
+        return result
+
+
+class NumberParameter(ToolParameter):
+    """Number parameter for tools."""
+
+    def __init__(
+        self,
+        description: str,
+        required: bool = False,
+        minimum: Optional[float] = None,
+        maximum: Optional[float] = None,
+    ):
+        super().__init__(description, required)
+        self.minimum = minimum
+        self.maximum = maximum
+
+    def to_dict(self) -> Dict[str, Any]:
+        result: Dict[str, Any] = {
+            "type": ParameterType.NUMBER,
+            "description": self.description,
+        }
+        if self.minimum is not None:
+            result["minimum"] = self.minimum
+        if self.maximum is not None:
+            result["maximum"] = self.maximum
+        return result
+
+
+class BooleanParameter(ToolParameter):
+    """Boolean parameter for tools."""
+
+    def to_dict(self) -> Dict[str, Any]:
+        return {"type": ParameterType.BOOLEAN, "description": self.description}
+
+
+class ObjectParameter(ToolParameter):
+    """Object parameter for tools."""
+
+    def __init__(
+        self,
+        description: str,
+        properties: Dict[str, ToolParameter],
+        required: bool = False,
+        required_properties: Optional[List[str]] = None,
+        additional_properties: bool = True,
+    ):
+        super().__init__(description, required)
+        self.properties = properties
+        self.required_properties = required_properties or []
+        self.additional_properties = additional_properties
+
+    def to_dict(self) -> Dict[str, Any]:
+        properties_dict: Dict[str, Any] = {}
+        for name, param in self.properties.items():
+            properties_dict[name] = param.to_dict()
+
+        result: Dict[str, Any] = {
+            "type": ParameterType.OBJECT,
+            "description": self.description,
+            "properties": properties_dict,
+        }
+
+        if self.required_properties:
+            result["required"] = self.required_properties
+
+        if not self.additional_properties:
+            result["additionalProperties"] = False
+
+        return result
+
+
+class ArrayParameter(ToolParameter):
+    """Array parameter for tools."""
+
+    def __init__(
+        self,
+        description: str,
+        items: ToolParameter,
```

Comment:
Shouldn't `items` be a `list[ToolParameter]`?

Reply:
Actually not, the JSON schema definition of the array item types is a type. See https://json-schema.org/understanding-json-schema/reference/array#items

Reply:
Ah ok, makes sense actually.